net/dnscache/dnscache.go

```diff
@@ -252,7 +250,26 @@ func (r *Resolver) lookupTimeoutForHost(host string) time.Duration {
 	return 10 * time.Second
 }
 
-func (r *Resolver) lookupIP(host string) (ip, ip6 net.IP, allIPs []net.IPAddr, err error) {
+func ip2addrs(n []net.IPAddr) []netip.Addr {
+	ips := make([]netip.Addr, 0, len(n))
+	for _, ip := range n {
+		if len(ip.IP) == 4 {
```

Comment:
docs say
> Note that in this documentation, referring to an IP address as an IPv4 address or an IPv6 address is a semantic property of the address, not just the length of the byte slice: a 16-byte slice can still be an IPv4 address.

Reply:
From reading the code this looks like it's referring to 4in6 addresses. I'm fine with converting 4in6 addresses to IPv4 addresses here. Wasn't sure if we wanted to preserve the 4in6 format for some reason.

Reply:
Updated to convert the address to IPv4 here.